Core/src/Material/SurfaceMaterialMapper.cpp

```diff
@@ -137,6 +138,31 @@ void Acts::SurfaceMaterialMapper::checkAndInsert(State& mState,
   }
 }
 
+void Acts::SurfaceMaterialMapper::collectMaterialVolume(
+    State& mState, const TrackingVolume& tVolume) const {
+  ACTS_VERBOSE("Checking volume '" << tVolume.volumeName()
+                                   << "' for material surfaces.")
+  ACTS_VERBOSE("- Insert Volume ...");
+  if (tVolume.volumeMaterial() != nullptr) {
+    mState.volumeMaterial[tVolume.geoID()] = tVolume.volumeMaterialSharedPtr();
+  }
+
+  // Step down into the sub volume
+  if (tVolume.confinedVolumes()) {
+    ACTS_VERBOSE("- Check children volume ...");
+    for (auto& sVolume : tVolume.confinedVolumes()->arrayObjects()) {
+      // Recursive call
+      collectMaterialVolume(mState, *sVolume);
```

Comment:
The surface and volume mapper both work the same way, they take a tracking geometry in input (with or with material) and return a material map. 
If we are doing volume mapping and surface material is already present we need to collect this surface material first to be able to add it to the material map we will return.
That way if you use a material map with surfaces in input it will return a full material map(surface+volume).